### PR TITLE
Depend on library projects rather than published artifacts

### PR DESCRIPTION
**Pull request by PaulKlauser**
This allows the sample apps to be used to test changes to the libraries without needing to publish

### CODE REVIEW

**Comment by PaulKlauser** [comment]
I found this to make contributing to this repo easier 😃  Let me know if there's an unintended side-effect to this that I'm not considering.

**Comment by DanielMartinus** [comment]
Hey @PaulKlauser, thanks for your contribution. Unfortunately this won't work since the api included library won't be included when building the konfetti-xml or konfetti-compose. Still have to work on a solution for that

File: konfetti/xml/build.gradle [hunk 1]
```diff
@@ -52,7 +52,7 @@ android {
 }
 
 dependencies {
-    api 'nl.dionsegijn:konfetti-core:2.0.1'
+    api project(':konfetti:core')
```
[the remark this answers is not in the document]
Comment (DanielMartinus) [reply konfetti/xml/build.gradle:55]: Unfortunately transitive dependencies don't work great for when building an `aar`. An aar only includes the files of that particular module and is able to use other modules to compile but unfortunately not able to expose files from other included transitive modules. Either you'll have to use fat aar to create your aar or make it an external dependency. For now I chose to make core an external dependency but I agree that the current setup is suboptimal. I'm going to dive into merging all the modules again and separate the internal code by package which seems to me for now the best way forward.

[the remark this answers is not in the document]
Comment (PaulKlauser) [reply konfetti/xml/build.gradle:55]: Ah right. Will bringing Compose into the library have any unintended side-effects for devs that aren't using Compose?

**Comment by PaulKlauser** [comment]
Closing, since this won't actually work when publishing the library's aar

**Comment by PaulKlauser** [comment]
@DanielMartinus it actually looks like maven-publish will generate the POM to point at the published dependency, even if the gradle file is pointed at the local project dependency. I can see this behavior by running `./gradlew konfetti:xml:generatePomFileForReleasePublication` and looking at the output POM file. 

Looks like the issue is just that the artifactId is named `core` instead of `konfetti-core`, which is what the core module is being published to so it ends up not matching. 

The solution would be to set the project name explicitly to `konfetti-core` which can be done by changing the directory name, or adding a line to the `settings.gradle`, which I'll add to this PR in a second

**Comment by DanielMartinus** [comment]
See #295 with a setup that works using the local core library and where publishing used the remote core dependency

**Pull request closed without merging**